lib/modules/manager/argocd/extract.ts

```diff
@@ -49,33 +53,48 @@ export function extractPackageFile(
   return deps.length ? { deps } : null;
 }
 
-function processSource(source: ApplicationSource): PackageDependency | null {
+function processSource(source: ApplicationSource): PackageDependency[] {
   // a chart variable is defined this is helm declaration
   if (source.chart) {
     // assume OCI helm chart if repoURL doesn't contain explicit protocol
     if (isOCIRegistry(source.repoURL) || !source.repoURL.includes('://')) {
       const registryURL = trimTrailingSlash(removeOCIPrefix(source.repoURL));
 
-      return {
-        depName: `${registryURL}/${source.chart}`,
-        currentValue: source.targetRevision,
-        datasource: DockerDatasource.id,
-      };
+      return [
+        {
+          depName: `${registryURL}/${source.chart}`,
+          currentValue: source.targetRevision,
+          datasource: DockerDatasource.id,
+        },
+      ];
     }
 
-    return {
-      depName: source.chart,
-      registryUrls: [source.repoURL],
+    return [
+      {
+        depName: source.chart,
+        registryUrls: [source.repoURL],
+        currentValue: source.targetRevision,
+        datasource: HelmDatasource.id,
+      },
+    ];
+  }
+
+  const dependencies: PackageDependency[] = [
+    {
+      depName: source.repoURL,
       currentValue: source.targetRevision,
-      datasource: HelmDatasource.id,
-    };
+      datasource: GitTagsDatasource.id,
+    },
+  ];
+
+  // Git repo is pointing to a Kustomize resources
+  if (source.kustomize?.images) {
```

Comment:
kustomize is only allowed with git repos? not allowed with chart?

Reply:
Not sure I understand: why and how would I use Kustomize from within a Helm Chart?

Reply:
I don't know, I'm not an argocd user

Reply:
For all I know: Yes, image overrides for Kustomize only make sense with Git, not with Chart.